### PR TITLE
wasmtime-cpp: add version 2.0.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **wasmtime-cpp/2.0.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wasmtime-cpp/all/conanfile.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from conan import ConanFile
+from conan import ConanFile, conan_version
```
Comment (prince-chrismc) [inline recipes/wasmtime-cpp/all/conanfile.py:1]:
```suggestion
from conan import ConanFile
```

File: recipes/wasmtime-cpp/all/conanfile.py [hunk 2]
```diff
@@ -47,23 +47,31 @@ def requirements(self):
     def package_id(self):
         self.info.clear()
 
+    @property
+    def _info(self):
+        return self if Version(conan_version).major < 2 else self.info
```
Comment (prince-chrismc) [inline recipes/wasmtime-cpp/all/conanfile.py:52]: Please remove, this is fixed in the next beta drop

**Review by prince-chrismc — COMMENTED** [review]
it's great just the changes in guidelines since this was open

File: recipes/wasmtime-cpp/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/wasmtime-cpp/all/conanfile.py]: Fixed

File: recipes/wasmtime-cpp/all/conanfile.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class WasmtimeCppConan(ConanFile):
     no_copy_source = True
 
     @property
-    def _minimum_cpp_standard(self):
+    def _min_cppstd(self):
```
Comment (uilianries) [inline recipes/wasmtime-cpp/all/conanfile.py:22]: Need to update it to templates. _min_cppstd is shorter and better.

**Review by uilianries — APPROVED** [review]

**Pull request merged**